core/src/Cache/MsalCacheItemBase.cs

```diff
@@ -0,0 +1,80 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+using System.Runtime.Serialization;
+using Microsoft.Identity.Core.Helpers;
+
+namespace Microsoft.Identity.Core.Cache
+{
+    [DataContract]
+    internal abstract class MsalCacheItemBase
+    {
+        [DataMember(Name = "unique_user_id", IsRequired = true)]
+        internal string UserIdentifier { get; set; }
+
+        [DataMember(Name = "environment", IsRequired = true)]
+        internal string Environment { get; set; }
+
+        [DataMember(Name = "client_info")]
+        internal string RawClientInfo { get; set; }
+
+        public ClientInfo ClientInfo { get; set; }
+
+        internal void InitClientInfo()
+        {
+            if (RawClientInfo != null)
+            {
+                ClientInfo = ClientInfo.CreateFromJson(RawClientInfo);
+            }
+        }
+
+        internal void InitRawClientInfoDerivedProperties()
```

Comment:
>InitRawClientInfoDerivedProperties [](start = 22, length = 34)

not a fan of someone external having to call this one...

Reply:
it is internal , not a part of api

Reply:
single responsibility. external code from the class is calling this, which is not good. If we really need this for what ever optimization, then it should be wrapped into a helper class that does this upon initialization or similar.

---
In reply to: [191035584](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1065#discussion_r191035584) [](ancestors = 191035584)